### PR TITLE
Resolves #26 Integrate eslint and prettier

### LINKED ISSUE #26: Add eslint and prettier to the project
In order to write quality code, we can use a tool called eslint to automatically check code styling. This helps to keep the codebase consistent.

https://dev.to/knowankit/setup-eslint-and-prettier-in-react-app-357b

Acceptance criteria:
1. Eslint and Prettier are configured.

### PR DESCRIPTION
**Pull request by anduxp**
## Note: 
The rules are exceedingly strict and cumbersome.

_Closes #26_

### CODE REVIEW

File: frontend/.eslintrc.json [hunk 1]
```diff
@@ -0,0 +1,28 @@
+{
+  "env": {
+    "browser": true,
+    "es2021": true
+  },
+  "extends": ["eslint:recommended", "plugin:react/recommended", "plugin:prettier/recommended"],
```
Comment (adamcorina) [inline frontend/.eslintrc.json:6]: extends is still the old one; check Cezar's

Comment (anduxp) [reply frontend/.eslintrc.json:6]: 'extends' was modified under package.json, as per Cezar's note, not under .eslintrc

File: frontend/package.json [hunk 2]
```diff
@@ -39,5 +34,17 @@
...
+  "devDependencies": {
+    "@typescript-eslint/eslint-plugin": "^5.30.7",
+    "@typescript-eslint/parser": "^5.30.7",
+    "eslint": "^8.20.0",
+    "eslint-config-prettier": "^8.5.0",
+    "eslint-plugin-prettier": "^4.2.1",
+    "eslint-plugin-react": "^7.30.1",
+    "prettier": "^2.7.1"
+  },
+  "lint": "eslint .",
```
Comment (code31415926535) [inline frontend/package.json:47]: `lint`, `lint:fix`, `format` should ho under `scripts` in order for them to be runnable.
Try running `npm run lint`.

File: frontend/tsconfig.json [hunk 3]
```diff
@@ -20,7 +17,6 @@
     "noEmit": true,
     "jsx": "react-jsx"
   },
-  "include": [
-    "src"
-  ]
+  "include": ["src"],
+  "exclude": ["node_modules", "build", "dist", "public", ".github"]
```
Comment (cezarderevlean) [inline frontend/tsconfig.json:21]: if we use `include`, we don't really need `exclude` too

[see why](https://www.typescriptlang.org/tsconfig#exclude)

File: frontend/package.json [hunk 4]
```diff
@@ -20,14 +20,12 @@
...
-    "extends": [
-      "react-app",
-      "react-app/jest"
-    ]
+    "eject": "react-scripts eject",
+    "lint": "eslint .",
+    "lint:fix": "eslint --fix",
+    "format": "prettier --write './**/*.{js,jsx,ts,tsx,css,md,json}' --config ./.prettierrc"
   },
+  "eslintConfig": {},
```
Comment (cezarderevlean) [inline frontend/package.json:28]: if we use `.eslintrc`, I'd remove the `eslintConfig` property from here altogether

but make sure that the changes you added to `.eslintrc` actually work as desired (both in terms of console errors/warnings and highlights/underlines from VSC) – one way to check this is to use double quotes for a string in a JS file.

**Review by cezarderevlean — CHANGES_REQUESTED** [review]

File: frontend/.prettierrc [hunk 5]
```diff
@@ -0,0 +1,16 @@
+{
+    "semi": true,
+    "tabWidth": 4,
+    "printWidth": 100,
+    "singleQuote": true,
+    "trailingComma": "all",
+    "jsxBracketSameLine": false,
```
Comment (cezarderevlean) [inline frontend/.prettierrc:7]: `jsxBracketSameLine` is [deprecated](https://prettier.io/docs/en/options.html#deprecated-jsx-brackets), please use `bracketSameLine`

and then run a format on `App.tsx` at least to see that it works and the bracket goes on the next line

File: frontend/src/App.tsx [hunk 6]
```diff
@@ -14,8 +14,7 @@ function App() {
           className="App-link"
           href="https://reactjs.org"
           target="_blank"
-          rel="noopener noreferrer"
-        >
+          rel="noopener noreferrer">
```
[the remark this answers is not in the document]
Comment (cezarderevlean) [reply frontend/src/App.tsx:17]: the bracket still is on the same line

File: frontend/tsconfig.json [hunk 7]
```diff
@@ -20,7 +17,6 @@
     "noEmit": true,
     "jsx": "react-jsx"
   },
-  "include": [
-    "src"
-  ]
+  "include": ["src"]
+  // "exclude": ["node_modules", "build", "dist", "public", ".github"]
```
Comment (cezarderevlean) [inline frontend/tsconfig.json:21]: doesn't this produce an error? as far as I know .json doesn't allow comments

**Review by cezarderevlean — CHANGES_REQUESTED** [review]

**Review by cezarderevlean — APPROVED** [review]

**Pull request merged**